sdk/tests/conformance2/reading/read-pixels-pack-parameters.html

```diff
@@ -116,57 +117,95 @@
-    debug("Testing width = " + width + ", height = " + height +
+    debug("Testing xoffset = " + xoffset + ", yoffset " + yoffset +
+          ", width = " + width + ", height = " + height +
           ", PACK_ALIGNMENT = " + packParams.alignment + ", PACK_ROW_LENGTH = " + packParams.rowLength +
           ", PACK_SKIP_PIXELS = " + packParams.skipPixels + " , PACK_SKIP_ROWS = " + packParams.skipRows);
     gl.pixelStorei(gl.PACK_ALIGNMENT, packParams.alignment);
     gl.pixelStorei(gl.PACK_ROW_LENGTH, packParams.rowLength);
     gl.pixelStorei(gl.PACK_SKIP_PIXELS, packParams.skipPixels);
     gl.pixelStorei(gl.PACK_SKIP_ROWS, packParams.skipRows);
-    wtu.glErrorShouldBe(gl, gl.NO_ERROR);
 
     var actualWidth = packParams.rowLength > 0 ? packParams.rowLength : width;
 
     var bytesPerPixel = 4; // see readPixels' parameters below, the format is gl.RGBA, type is gl.UNSIGNED_BYTE
     var padding = calculatePaddingBytes(bytesPerPixel, packParams.alignment, actualWidth);
+    var bytesPerRow = actualWidth * bytesPerPixel + padding;
 
-    var size = bytesPerPixel * actualWidth * height + padding * (height - 1);
-    if (size < 0)
-        size = 0;
-    var skipSize = (packParams.skipPixels * bytesPerPixel +
-                    packParams.skipRows * (bytesPerPixel * actualWidth + padding));
-    var array;
+    var size = bytesPerRow * (height - 1) + bytesPerPixel * width;
+    var skipSize = packParams.skipPixels * bytesPerPixel + packParams.skipRows * bytesPerRow;
+    var array = new Uint8Array(skipSize + size);
+    for (var ii = 0; ii < skipSize + size; ++ii) {
+        array[ii] = clearColor[ii % bytesPerPixel];
+    }
     if (usePixelPackBuffer) {
         var buffer = gl.createBuffer();
         gl.bindBuffer(gl.PIXEL_PACK_BUFFER, buffer);
-        gl.bufferData(gl.PIXEL_PACK_BUFFER, size + skipSize, gl.STATIC_DRAW);
+        gl.bufferData(gl.PIXEL_PACK_BUFFER, array, gl.STATIC_DRAW);
         var offset = 0;
-        gl.readPixels(0, 0, width, height, gl.RGBA, gl.UNSIGNED_BYTE, offset);
+        gl.readPixels(xoffset, yoffset, width, height, gl.RGBA, gl.UNSIGNED_BYTE, offset);
     } else {
-        array = new Uint8Array(size + skipSize);
-        gl.readPixels(0, 0, width, height, gl.RGBA, gl.UNSIGNED_BYTE, array);
+        gl.readPixels(xoffset, yoffset, width, height, gl.RGBA, gl.UNSIGNED_BYTE, array);
     }
-    wtu.glErrorShouldBe(gl, gl.NO_ERROR);
-
-    if (!size)
+    if (size == 0)
         return;
 
     if (usePixelPackBuffer) {
-        var arrBuffer = new ArrayBuffer(size);
-        gl.getBufferSubData(gl.PIXEL_PACK_BUFFER, skipSize, arrBuffer);
-        var array = new Uint8Array(arrBuffer);
+        var arrBuffer = new ArrayBuffer(skipSize + size);
+        gl.getBufferSubData(gl.PIXEL_PACK_BUFFER, 0, arrBuffer);
+        array = new Uint8Array(arrBuffer);
     }
 
+    // Check skipped bytes are unchanged.
+    for (var ii = 0; ii < skipSize; ++ii) {
+        if (array[ii] != clearColor[ii % bytesPerPixel]) {
```

Comment:
Ah, I see that this is set in JavaScript rather than drawn via OpenGL, so the answer is "no".